src/LinuxTracing/TracerImpl.cpp

```diff
@@ -72,22 +72,26 @@ TracerImpl::TracerImpl(
       listener_{listener} {
   CHECK(listener_ != nullptr);
 
-  if (unwinding_method_ != CaptureOptions::kUndefined) {
-    uint32_t stack_dump_size = capture_options.stack_dump_size();
-    if (stack_dump_size > kMaxStackSampleUserSize || stack_dump_size == 0) {
-      ERROR("Invalid sample stack dump size: %u; Reassigning to default: %u", stack_dump_size,
-            kMaxStackSampleUserSize);
+  uint32_t stack_dump_size = capture_options.stack_dump_size();
+  if (stack_dump_size == std::numeric_limits<uint16_t>::max()) {
+    if (unwinding_method_ == CaptureOptions::kDwarf) {
+      LOG("No sample stack dump size was set; Assigning to default: %u", kMaxStackSampleUserSize);
       stack_dump_size = kMaxStackSampleUserSize;
+    } else if (unwinding_method_ == CaptureOptions::kFramePointers) {
+      stack_dump_size = kMaxStackSampleUserSizeFramePointer;
     }
-    stack_dump_size_ = static_cast<uint16_t>(stack_dump_size);
-    std::optional<uint64_t> sampling_period_ns =
-        ComputeSamplingPeriodNs(capture_options.samples_per_second());
-    FAIL_IF(!sampling_period_ns.has_value(), "Invalid sampling rate: %.1f",
-            capture_options.samples_per_second());
-    sampling_period_ns_ = sampling_period_ns.value();
-  } else {
-    sampling_period_ns_ = 0;
+  } else if (stack_dump_size > kMaxStackSampleUserSize || stack_dump_size == 0) {
+    ERROR("Invalid sample stack dump size: %u; Reassigning to default: %u", stack_dump_size,
+          kMaxStackSampleUserSize);
+    stack_dump_size = kMaxStackSampleUserSize;
   }
+  stack_dump_size_ = static_cast<uint16_t>(stack_dump_size);
+  std::optional<uint64_t> sampling_period_ns =
+      ComputeSamplingPeriodNs(capture_options.samples_per_second());
+  if (!sampling_period_ns.has_value()) {
+    sampling_period_ns = 0;
```

Comment:
I don't really like using 0 like that, as no sampling is theoretically infinite period. So either define a constant (with either 0 or maybe better the maximum value of `uint64_t`) or (probably overkill) make `sampling_period_ns_` `optional`.